### DISCUSSION TITLE
Images not loading in NextJS Headless WordPress demo fresh install

### DISCUSSION THREAD

**Discussion by MattBonacini** [vercel/next.js#39615] [Help] [answered]
Hi everyone! I'm trying NextJS for the first time. My goal is to use it as front-end for an headless WordPress website.

I tried to deploy this template: https://vercel.com/templates/next.js/isr-blog-nextjs-wordpress

I just followed two simple steps: 
1. Click on "Deploy"
2. Specified my GraphQL url from my WP install. I checked on my WP install and GraphQL seems to work there. URL is https://wordpress-406795-2818064.cloudwaysapps.com/graphql

It seems images are not loaded: https://isr-blog-nextjs-wordpress.vercel.app/

I try to do some searches but it seems I can't find a solution. What did I do wrong? It seems it gets the rest of the content but just skips images.

EDIT: some images do work, but only inside content, not when they're set as featured image: https://wordpress-advisor.vercel.app/posts/dolorem-blanditiis-adipisci-nihil-ut-in

EDIT 2: by the way, got the same config locally, with Local by Flywheel and featured images work without issues there.

**Comment by bobarros** [comment, marked as answer]
Did you add the env variable mentioned here? 

https://github.com/vercel/next.js/blob/canary/examples/cms-wordpress/next.config.js

**Reply by MattBonacini** [reply]
Wow thank you for the super fast reply. That was it. I've realized only now that .env is ignored in git. So of course in my repo there wasn't a reference to that variable.

I've created a .env file that I committed to the repository, with the graphql url. For now this is the only env variable I have. Is this good practice or should I never commit .env files?

**Reply by gaston-flores** [reply]
It's generally a bad idea, since `.env` files are used to store secrets and sensitive things you don't really want in version control. If you are using vercel you can [add variables to the environment there](https://vercel.com/docs/concepts/projects/environment-variables).

**Reply by MattBonacini** [reply]
Awesome, I'll do that, thanks!
